packages/discovery-react-components/src/components/SearchFacets/SearchFacets.tsx

```diff
@@ -101,19 +114,34 @@ const SearchFacets: FC<SearchFacetsProps> = ({
         label: collection.name || ''
       };
     });
+
   const [collectionSelectionState, setCollectionSelectionState] = useState<
     SelectedCollectionItems['selectedItems']
   >(initialSelectedCollections);
+
+  const [fetchState, setFetchState] = useState<'init' | 'loading' | 'success' | 'error'>('init');
+
   const { fetchAggregations, performSearch } = useContext(SearchApi);
   const aggregations = aggregationResults || [];
   const mergedMessages = { ...defaultMessages, ...messages };
+
   const componentSettingsAggregations =
     overrideComponentSettingsAggregations ||
     (componentSettings && componentSettings.aggregations) ||
     [];
 
   useEffect(() => {
-    fetchAggregations(searchParameters);
+    async function fetchData() {
+      setFetchState('loading');
+      try {
+        await fetchAggregations(searchParameters);
+        setFetchState('success');
+      } catch (error) {
+        setFetchState('error');
+      }
+    }
+
+    fetchData();
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
No, because the code is using `searchParameters`, but we only want to re-run the `useEffect` if `searchParameters.aggregation` changes.

Reply:
I think since you didn't add this comment, we can leave it for this PR. But that is not a good reason. There are other ways to run `useEffect` when a value changes without leaving out necessary dependencies. We should address this in a follow up PR.

Reply:
Do you have a proposal in mind? We could depend on `searchParameters`, but then we'd have to use `useDeepCompareEffect`, in order to do a deep compare of the object. Then inside, we can check if `searchParameters.aggregation` has changed before fetching -- but that requires keeping the previous value in state.

Reply:
Using [`usePrevious`](https://usehooks.com/usePrevious/) you can keep track if a specific change.

```ts
const aggregationsChanged = usePrevious(searchParameters.aggregation);

useEffect(() => {
  async function fetchData() {
    setFetchState('loading');
    try {
      await fetchAggregations(searchParameters);
      setFetchState('success');
    } catch (error) {
      setFetchState('error');
    }
  }

  if (aggregationsChanged) {
    fetchData();
  }
}, [fetchAggregations, searchParameters, aggregationsChanged]);
```

Reply:
Right. Although, `usePrevious` returns the previous value, so you'd have to do a compare against current value inside the `useEffect`:

```javascript
const prevSearchParamsAggregation = usePrevious(searchParameters.aggregation);

useEffect(() => {
  async function fetchData() {
    setFetchState('loading');
    try {
      await fetchAggregations(searchParameters);
      setFetchState('success');
    } catch (error) {
      setFetchState('error');
    }
  }

  if (prevSearchParamsAggregation !== searchParameters.aggregation) {
    fetchData();
  }
}, [fetchAggregations, setFetchState, searchParameters, prevSearchParamsAggregation]);
```

Reply:
Whoops! My bad, i was thinking of the `useCompare` hook that we have in tooling which tells us if it changed. You are correct, `usePrevious` gives the previous value.

You can check it outside of the `useEffect` rather than passing everything in as a dependency or make another hook similar to what we have in tooling.

```ts
export const useCompare = val => {
  const prevVal = usePrevious(val);
  return !isEqual(prevVal, val);
};
```

then the code i posted above would just be changed to look like:
```ts
const aggregationsChanged = useCompare(searchParameters.aggregation);

useEffect(() => {
  async function fetchData() {
    setFetchState('loading');
    try {
      await fetchAggregations(searchParameters);
      setFetchState('success');
    } catch (error) {
      setFetchState('error');
    }
  }

  if (aggregationsChanged) {
    fetchData();
  }
}, [fetchAggregations, setFetchState, searchParameters, aggregationsChanged]);
```

Reply:
Updated to use `useCompare`.